introspection-engine/connectors/mongodb-introspection-connector/src/sampler.rs

```diff
@@ -30,6 +30,8 @@ pub(super) async fn sample(database: Database) -> crate::Result<IntrospectionRes
         let options = AggregateOptions::builder().allow_disk_use(Some(true)).build();
 
         let mut documents = collection
+            // Any update to this number size must be done there as well:
+            // https://github.com/prisma/prisma-engines/blob/master/introspection-engine/connectors/mongodb-introspection-connector/src/sampler/statistics.rs#L118
```

Comment:
We can define that as a constant instead: in this file, something like

```rust
pub(super) const SAMPLE_SIZE: usize = 1000;
```

and in `statistics.rs`:

```rust
                Some(format!(
                    "Multiple data types found {} out of {} sampled entries",
                    percentages,
                    sample size
                ))
```

We're probably going to make sampling smarter/more configurable at some point and the sample size (and possibly description) will have to be dynamic, but we'll change that when we get ther.

Reply:
usize says `the trait `From<usize>` is not implemented for `Bson`` so I switched to u8

Reply:
Yep, `i16`, `i32`, `i64` or `isize` should work.

Reply:
@Jolg42 heads up that you pushed `u32` and not `i32` (in case it wasn't intended and there isn't a specific reason to use an unsigned integer). Apparently that's supported as well, and there's even a special case for the lower half of `u32` (https://github.com/mongodb/bson-rust/blob/785ae7c171f36342b9e30f59ea37ce50f7c8eebb/src/bson.rs#L296), so it won't be converted to `i64` as I would assume, but you may save a few CPU cycles by using `i32` directly 😄